clarity/src/vm/analysis/type_checker/natives/mod.rs

```diff
@@ -248,9 +248,22 @@ fn check_special_let(
             checker,
             typed_result.type_size()?,
         )?;
-        out_context
-            .variable_types
-            .insert(var_name.clone(), typed_result);
+        // Beginning in Clarity 2, traits can be propagated into let variables.
+        if checker.clarity_version >= ClarityVersion::Clarity2 {
```

Comment:
I see this feature gate repeated a few times.  Do you think we should instead have a method that takes the type signature and Clarity version as arguments and either inserts the typed result into the `traits_references` or `variable_types` maps, based on the version?

Reply:
Refactored in c6eaf08cb

Reply:
I'm not seeing the refactoring in that commit?

Reply:
Ah, sorry, it seems that earlier one was for a different comment. Now this is refactored in 17e16ea2b.